src/assets/scripts/client/base/BaseModel.js

```diff
@@ -45,4 +45,27 @@ export default class BaseModel {
     reset() {
         throw new TypeError('BaseModel#reset method must be implemented by the class extending BaseModel');
     }
+
+    /**
+     * This will verify if the given argument is a string and will return a modified pre-value for the _id
+     *
+     * @for BaseModel
+     * @method veriftyModelName
+     * @param {string / Object}
+     */
+    verifyModelName(modelName) {
+        if (typeof modelName === 'string') {
+            return `${modelName}-`;
+        }
+
+        // This will give more infomation on airpot names and runways to help identify if there is an issue.
+        if (modelName.name === undefined && typeof modelName === 'object') {
```

Comment:
This will show additional information on the object that is used to create it give you `_id`  values like 
`BaseModel-Caribbean Airlines-1087`  So you can see if it is only happening with certain Data, But I can just remove it.
I did an oversimplification when i was cleaning up the logic here
This would be a more accurate version
```
if (modelName.name === undefined) {
            modelName.name = '';
        }

        // This will give more information on airport names and runways to help identify if there is an issue.
        if (typeof modelName === 'object') {
            console.log(modelName);
            return `BaseModel-${modelName.name}-`;
        }
```